internals/cli/masker/stream.go

```diff
@@ -64,31 +65,29 @@ func (s *stream) flush(n int) error {
 
 		if exists {
 			// Get any unprocessed bytes before this match to the destination.
-			beforeMatch := s.buf.upToIndex(i)
-
-			_, err := s.dest.Write(beforeMatch)
+			bytesBeforeMatch, err := s.buf.writeUpToIndex(s.dest, i)
 			if err != nil {
 				return err
 			}
 
 			// Only write the redaction text if there were bytes between this match and the previous match
 			// or this is the first flush for the buffer.
-			if len(beforeMatch) > 0 || s.buf.currentIndex == 0 {
+			if bytesBeforeMatch > 0 || s.buf.currentIndex == 0 {
 				_, err = s.dest.Write([]byte("<redacted by SecretHub>"))
 				if err != nil {
 					return err
 				}
 			}
 
 			// Drop all bytes until the end of the mask.
-			_ = s.buf.upToIndex(i + int64(length))
+			_, _ = s.buf.writeUpToIndex(ioutil.Discard, i+int64(length))
```

Comment:
In the current setup `writeUpToIndex` can never return an error here, but the function signature does not enforce this. Since `flush()` does have `error` as a return type, would it be a problem if we just forward the error? That way we prevent any bugs when `writeUpToIndex` is changed so that it can return when a `ioutil.Discarder` is provided as an argument.

Reply:
Yes, good point. This seems like a place where we could easily introduce a bug by mistake later on unless we handle the error.

Reply:
I added the error handling.

Reply:
:ok_hand: